client/Client.cpp

```diff
@@ -558,7 +558,9 @@ int CClient::sendRequest(const CPackForServer * request, PlayerColor player)
 
 void CClient::battleStarted(const BattleInfo * info)
 {
+	BattleInfo::adjustOppositeBonuses(const_cast<BattleInfo *>(info));

```

Comment:
I think not worth it. At least this is how determined where info can be changed, and where it shouldn't be changed.
More cases when it may be changed increase probability of errors.

Reply:
I still think adjustOppositeBonuses can be an instance method and it can const_cast(this) inside to make required changes. This will be kind of incapsulation, BattleInfo decides that such change does not affect battle itself and can be treated as const. But I still want to take a bit more closer look onto all it.

Reply:
It's called from static setupBattle - that's why it's static.
adjustOppositeBonuses receives non-const  argument and should not do const_cast inside!